File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        left, right, res = 0, len(mat)-1, 0
        for r in range(0, len(mat)):
            if left == right:
                res += mat[r][left]
            else:
                res += mat[r][left] + mat[r][right]
            left += 1
            right -= 1
        return res

File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.kt
class Solution {
    fun diagonalSum(mat: Array<IntArray>): Int {
        var res = 0
        var left = 0
        var right = mat.size-1
        for (r in 0..(mat.size-1)) {
            if (left === right) {
                res += mat[r][left]
            } else {
                res += mat[r][left] + mat[r][right]
            }
            left++
            right--
        }
        return res
    }
}